src/server/pfs/server/trigger.go

```diff
@@ -131,16 +131,15 @@ func (d *driver) isTriggered(txnCtx *txncontext.TransactionContext, t *pfs.Trigg
 		ci := newHead
 		var commits int64
 		for commits < t.Commits {
```

Comment:
I think it's a bit cleaner to write this loop like

```Go
for ci.ParentCommit != nil && commits < t.Commits {
			if ci.Commit.Id == oldHead.Commit.Id {
				break
			}
			var err error
			ci, err = d.resolveCommit(ctx, txnCtx.SqlTx, ci.ParentCommit)
			if err != nil {
				return false, err
			}
			commits++
		}
```

Reply:
To me that seems less clear, but that may just be how I read it. I read it as traverse t.Commits, but break out if the oldHead or empty commit we create is reached. What you proposed just pulls one of the break conditions up to the loop conditional. It seems like you might as well put the full conditional there if you do that.